qt/python/mantidqt/project/test/test_project.py

```diff
@@ -247,6 +245,44 @@ def test_workspace_groups_are_not_duplicated_when_saving(self, saver):
                                  plots_to_save="mocked_figs",
                                  interfaces_to_save="mocked_interfaces")
 
+    def test_filter_unaltered_workspaces_function_removes_workspaces_that_have_only_been_loaded(self):
```

Comment:
Thanks for adding these thorough tests. I'm wondering if there is any way we could remove all of the calls to mantid algorithms and instead use mock ADS and mock workspace objects? Any tests with algorithms and files keep adding that extra seconds for all of the tests to run and slow things down generally. I'd ideally like to get rid of `CreateSampleWorkspace` etc as developers tend to copy of other tests and things get incrementally slower.